pkg/controller/endpointslice/reconciler.go

```diff
@@ -56,12 +57,62 @@ type endpointMeta struct {
 // slices for the given service. It creates, updates, or deletes endpoint slices
 // to ensure the desired set of pods are represented by endpoint slices.
 func (r *reconciler) reconcile(service *corev1.Service, pods []*corev1.Pod, existingSlices []*discovery.EndpointSlice, triggerTime time.Time) error {
-	addressType := discovery.AddressTypeIPv4
+	slicesToDelete := []*discovery.EndpointSlice{}                                    // slices that are no longer  matching any address the service has
+	errs := []error{}                                                                 // all errors generated in the process of reconciling
+	slicesByAddressType := make(map[discovery.AddressType][]*discovery.EndpointSlice) // slices by address type
 
-	if endpointutil.IsIPv6Service(service) {
-		addressType = discovery.AddressTypeIPv6
+	// addresses that this service supports [o(1) find]
+	serviceSupportedAddresses := getAddressTypesForService(service)
```

Comment:
If this returned a `map[family]bool` code would be easier to read

Reply:
it was like that. Then we changed to struct{} because `bool` allocates unneeded mem. `struct{}` does not. we really don't care for value, we care for the key presence.

Reply:
My point was that bool means callers can say `if serviceSupportedAddresses[family] { ` instead of `if _, found := serviceSupportedAddresses[family]; found {`

This case is minor.